examples/controlnet/train.sh

```diff
@@ -1,10 +1,29 @@
 ###controlnet.sh PSEUDO###
-env_name=${1}
-model_name=${2}
-batch_size=${3}
-output_dir=${4:-"outputs"}
+if [[ "$@" == *"--task"* ]]; then # --env_name: conda env name
+  env_name="$(echo "$@" | sed -n 's/.*--task \([^ ]*\).*/\1/p')"
+  echo "env_name: $env_name"
+fi
+
+if [[ "$@" == *"--model_name"* ]]; then
+  model_name="$(echo "$@" | sed -n 's/.*--model_name \([^ ]*\).*/\1/p')"
+fi
+
+if [[ "$@" == *"--batch_size"* ]]; then 
+  batch_size="$(echo "$@" | sed -n 's/.*--batch_size \([^ ]*\).*/\1/p')"
+fi
+
+output_dir=outputs
+if [[ "$@" == *"--output_dir"* ]]; then 
+  output_dir="$(echo "$@" | sed -n 's/.*--output_dir \([^ ]*\).*/\1/p')"
+fi
+
```

Comment:
This method of parsing arguments looks pretty cumbersome. Have you considered using short args (like `-m`, `-t`, `-b`) and parsing them using `getopts` like [this](https://github.com/moreh-dev/transformers/blob/8428265f1879d34b56e2f24387ee898f2b79da0a/examples/pytorch/language-modeling/train_clm.sh#L1)?

Reply:
Thanks. I tested out the method you recommended, and it worked well. The script looks much neater now